source/Modules/StepBro.ExecutionHelper.Access/Access.cs

```diff
@@ -0,0 +1,198 @@
+﻿using StepBro.Core.IPC;
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Threading;
+
+namespace StepBro.ExecutionHelper
+{
+    public class Access
+    {
+        private Pipe m_executionHelperPipe = null;
+        bool m_executionHelperStarted = false;
+        bool m_closeWhenExecutionHelperCloses = false;
+        EventHandler<Tuple<string, string>> receivedDataEventHandler = null;
+
+        private void ReceivedData(Tuple<string, string> message)
+        {
+            if (message.Item1 == nameof(StepBro.ExecutionHelper.Messages.ShortCommand))
+            {
+                var cmd = JsonSerializer.Deserialize<StepBro.ExecutionHelper.Messages.ShortCommand>(message.Item2);
+                if (cmd == StepBro.ExecutionHelper.Messages.ShortCommand.Close && m_closeWhenExecutionHelperCloses)
+                {
+                    Thread.Sleep(100);
+                    // m_executionHelperPipe.Dispose();
+                    System.Environment.Exit(0); // Close the application gracefully
+                }
+            }
+        }
+
+        public bool CreateExecutionHelper(bool closeWhenExecutionHelperCloses = false)
+        {
+            m_closeWhenExecutionHelperCloses = closeWhenExecutionHelperCloses;
+            bool result = true;
+
+            var hThis = GetConsoleWindow();
+
+            string path = Assembly.GetExecutingAssembly().Location;
+            var folder = Path.GetDirectoryName(path);
+
+            string pipename = hThis.ToString("X");
+            m_executionHelperPipe = Pipe.StartServer("StepBroExecutionHelper", pipename);
+            var executionHelper = new System.Diagnostics.Process();
+            executionHelper.StartInfo.FileName = Path.Combine(folder, "../StepBro.ExecutionHelper.exe"); //../ because ExecutionHelper is in the main bin folder and this is the Modules folder
+            executionHelper.StartInfo.Arguments = pipename;
+            m_executionHelperStarted = executionHelper.Start();
```

Comment:
Maybe the helper is started already, so try to connect first (as slave) and then start Helper app if it is not possible to connect.